ionoscloud/resource_server.go

```diff
@@ -1509,6 +1520,8 @@ func setResourceServerData(ctx context.Context, client *ionoscloud.APIClient, d
 			if err := d.Set("boot_cdrom", *server.Properties.BootCdrom.Id); err != nil {
 				return fmt.Errorf("error setting boot_cdrom %w", err)
 			}
+		} else {
```

Comment:
@cbeti-ionos why are we setting this to `nil` here?

Reply:
The BootCdrom property of the resource can currently only be nulled from the backend side.
One way to do this is when setting a volume as primary boot device - which in turn overrides the cdrom and sets this property to null.
This is why we're setting it to nil here, in the reader of the resource.